### PR TITLE
Update StarSpawn to StarCloudData

### PR DESCRIPTION
**Pull request by Tmcg2**
Add and better specify the struct used to control how stars and
 clouds are handled in various courses.

### CODE REVIEW

File: include/objects.h [hunk 1]
```diff
@@ -402,6 +395,22 @@ extern s32 gLeafParticle[];
...
+    // See `func_800788F8` to see how rotY is used to decide whether and where to display clouds/stars
+    // See `func_80078A44` and `func_800789AC` to see how stars and clouds (respectively) are looped over
+    /* 0x00 */ u16 rotY;
+    /* 0x02 */ u16 posY;
+    // Can be bigger than 100!
+    /* 0x04 */ u16 scalePercent;
+    // There are a couple different cloud shapes, this decides which one is used
+    // For stars, this is always 0 since they don't have multiple textures
+    /* 0x06 */ u16 subType;
+} StarCloudData; // size = 0x8
```
Comment (coco875) [inline include/objects.h:412]: why not make two struct even if they are equivalent to see more difference between clound and star
```suggestion
} StarData, CloundData; // size = 0x8
```

File: src/code_8006E9C0.c [hunk 2]
```diff
@@ -642,34 +643,34 @@ void func_80070328(StarSpawn *arg0) {
...
  * They are instead objects that seemingly hover in the air around the player
  * They have no true x/y/z position, instead they seem to be kept in a position relative to the
- * player they hang around. There is however an x/y position for where they should be on screen
+ * player they hang around. There is however an y rotation and y position for where they should be on screen
  * when they are visbile (unk_09E[0] and [1]).
  * sizeScaling is some sort of size scaling on the start texture.
  * unk_0A2 is an alpha value, used to make the star twinkle.
 **/
-void func_800703E0(s32 objectIndex, s32 arg1, StarSpawn *arg2) {
+void func_800703E0(s32 objectIndex, s32 arg1, StarCloudData *arg2) {
```
Comment (coco875) [inline src/code_8006E9C0.c:651]: same here but for star

File: include/objects.h (hunk 1 above)
Comment (Tmcg2) [reply include/objects.h:412]: I didn't realize you could do that in C, nifty.

**Review by MegaMech — COMMENTED** [review]
lgtm and Coco's suggestions seems reasonable. I only glanced at the PR though.

**Comment by Tmcg2** [comment]
Updated based on Coco's comments. Typedef'ing multiple structs types at once is a neat feature I didn't know about, so thank you for the pointer!

**Review by MegaMech — DISMISSED** [review]

**Comment by Tmcg2** [comment]
Rebased onto master to fix the merge conflicts

**Pull request merged**